### PR TITLE
Issue #26: Fix splices in requires clause

### PR DESCRIPTION
**Pull request by delimbetov**
*Issue number of the reported bug or feature request: #26*

**Describe your changes**
Fixed parsing of splices in requires clauses

**Testing performed**
Added tests for the 4 requirements types: simple, type, compound and nested.

### CODE REVIEW

File: clang/test/Reflection/splice-types.cpp [hunk 1]
```diff
@@ -225,3 +237,146 @@ struct T : [:Rs:]... {};
 using A = T<^B1, ^B2, ^B3>;
 static_assert(A::value1 + A::value2 + A::value3 == 6);
 }  // namespace base_class_specifiers
+
```
Comment (katzdm) [inline clang/test/Reflection/splice-types.cpp:240]: nit: Would you mind adding a new centered comment header here?

```
// ===============
// requires_clause
// ===============
```

File: clang/test/Reflection/splice-types.cpp [hunk 2]
```diff
@@ -225,3 +237,146 @@ struct T : [:Rs:]... {};
...
+  requires same_as<T, [:^Namespace:]::Addable>;
+};
+
+static_assert(nested_addable<Addable>);
+static_assert(!nested_addable<NonAddable>);
+
+static_assert(nested_addable2<Addable>);
+static_assert(!nested_addable2<NonAddable>);
+
+}
```
Comment (katzdm) [inline clang/test/Reflection/splice-types.cpp]: nit:

```cpp
}  // namespace requires_clause
```

**Review by katzdm — COMMENTED** [review]
Looks amazing! Two very minor comment nits, and we're good to go.

File: clang/test/Reflection/splice-types.cpp (hunk 1 above)
Comment (delimbetov) [reply clang/test/Reflection/splice-types.cpp:240]: done

File: clang/test/Reflection/splice-types.cpp (hunk 2 above)
Comment (delimbetov) [reply clang/test/Reflection/splice-types.cpp]: done

**Pull request merged**